### PR TITLE
Standard Config Load

### PR DESCRIPTION
**Pull request by hexonaut**
Standardize the environment variables to use when loading configs. Default is to look in the local `script` directory, but can override with arbitrary json with the `_TEXT` env variant.

### CODE REVIEW

File: src/ScriptTools.sol [hunk 1]
```diff
@@ -37,6 +37,18 @@ library ScriptTools {
...
+    
+    /**
+     * @notice Use standard environment variables to load config.
+     * @dev Will first check FOUNDRY_SCRIPT_CONFIG_TEXT for raw json text.
+     *      Falls back to FOUNDRY_SCRIPT_CONFIG for a standard file definition.
+     */
+    function loadConfig() internal returns (string memory config) {
+        config = vm.envOr("FOUNDRY_SCRIPT_CONFIG_TEXT", string(""));
+        if (eq(config, "")) {
+            config = readInput(vm.envString("FOUNDRY_SCRIPT_CONFIG"));
```
Comment (gbalabasquer) [inline src/ScriptTools.sol:49]: Should we pass a string for a second fallback?

Comment (hexonaut) [reply src/ScriptTools.sol:49]: Like `ScriptTools.loadConfig("fallback")` which would attempt to load the json file "fallback.json" if no env var? If so then yes makes sense to me.

Comment (hexonaut) [reply src/ScriptTools.sol:49]: Anyways. I've added it.

Comment (gbalabasquer) [reply src/ScriptTools.sol:49]: Yeah exactly this, but I'm not seeing it

Comment (gbalabasquer) [reply src/ScriptTools.sol:49]: Probably missing the push

Comment (hexonaut) [reply src/ScriptTools.sol]: ok should be good now

File: src/ScriptTools.sol [hunk 2]
```diff
@@ -40,6 +40,19 @@ library ScriptTools {
...
+     * @notice Use standard environment variables to load config.
+     * @dev Will first check FOUNDRY_SCRIPT_CONFIG_TEXT for raw json text.
+     *      Falls back to FOUNDRY_SCRIPT_CONFIG for a standard file definition.
+     *      Finally will fall back to the given string.
+     */
+    function loadConfig(string memory name) internal returns (string memory config) {
+        config = vm.envOr("FOUNDRY_SCRIPT_CONFIG_TEXT", string(""));
+        if (eq(config, "")) {
+            config = readInput(vm.envOr("FOUNDRY_SCRIPT_CONFIG", name));
+        }
```
Comment (gbalabasquer) [inline src/ScriptTools.sol:54]: Would the following work?

```suggestion
        config = vm.envOr("FOUNDRY_SCRIPT_CONFIG_TEXT", readInput(vm.envOr("FOUNDRY_SCRIPT_CONFIG", name)));
```

Comment (hexonaut) [reply src/ScriptTools.sol:54]: Yes this should work. The issue before was that the second part would be evaluated which reverted before it was converted into the optional version. This should work in the one line.

**Review by gbalabasquer — APPROVED** [review]
LGTM

**Pull request merged**